File: exercism/lua/difference-of-squares/difference-of-squares.lua
local function square_of_sum(n)
  local total = 0
  for i=1, n do
    total = total + i
  end
  return math.pow(total, 2)
end

local function sum_of_squares(n)
  local total = 0
  for i=1, n do
    total = total + math.pow(i, 2)
  end
  return total
end

local function difference_of_squares(n)
  return square_of_sum(n) - sum_of_squares(n)
end

return {
  square_of_sum = square_of_sum,
  sum_of_squares = sum_of_squares,
  difference_of_squares = difference_of_squares
}

File: exercism/go/difference-of-squares/difference_of_squares.go
package diffsquares

import "math"

// SquareOfSums returns the square of a sum for a given range
func SquareOfSums(r int) int {
	sum := 0
	for i := 1; i <= r; i++ {
		sum += i
	}
	return int(math.Pow(float64(sum), 2.0))
}

// SumOfSquares returns a sum of square for a given range
func SumOfSquares(r int) int {
	sum := 0
	for i := 1; i <= r; i++ {
		sum += int(math.Pow(float64(i), 2.0))
	}
	return sum
}

// Difference returns a difference between a SumOfSquares and a SquareOfSums
func Difference(r int) int {
	sumOfSquares := SumOfSquares(r)
	squareOfSums := SquareOfSums(r)

	return squareOfSums - sumOfSquares
}
